src/content/en/fundamentals/engage-and-retain/push-notifications/notification-behaviour.md

```diff
@@ -0,0 +1,255 @@
+
+### Notification Click Event
+
+When a user clicks on a notification the default behaviour is for nothing
+to happen, it doesn't even close / remove the notification.
+
+The common practice for a notification click is for it to close and perform some other logic
+(i.e. open a window or make some API call to the application).
+
+To achieve this we need to add a 'notificationclick' event listener to our service worker. This
+will be called when ever a notification is clicked.
+
+    self.addEventListener('notificationclick', function(event) {
+      const clickedNotification = event.notification;
+      clickedNotification.close();
+
+      // Do something as the result of the notification click
+      const promiseChain = doSomething();
+      event.waitUntil(promiseChain);
+    });
+
+As you can see in this example, the notification was clicked can be accessed via the
+`event.notification parameter`. From this we can via the properties on the notification, in
+this case we call it's `close()` method and then we are free to perform any task we wish in the
+background like a normal event.
+
+Note: You still need to make use of event.waitUntil() to keep the service worker running
+while your code is busy.
+
+### Actions
+
+Actions allow you to give users another level of interaction with your users
+over just clicking the notification.
+
+In the previous section you saw how to define actions when calling
+`showNotification()`:
+
+        const title = 'Actions Notification';
+        const options = {
+          actions: [
+            {
+              action: 'coffee-action',
+              title: 'Coffee',
+              icon: '/images/demos/action-1-128x128.png'
+            },
+            {
+              action: 'doughnut-action',
+              title: 'Doughnut',
+              icon: '/images/demos/action-2-128x128.png'
+            },
+            {
+              action: 'gramophone-action',
+              title: 'gramophone',
+              icon: '/images/demos/action-3-128x128.png'
+            },
+            {
+              action: 'atom-action',
+              title: 'Atom',
+              icon: '/images/demos/action-4-128x128.png'
+            }
+          ]
+        };
+
+        const maxVisibleActions = Notification.maxActions;
+        if (maxVisibleActions < 4) {
+          options.body = `This notification will only display ` +
+            `${maxVisibleActions} actions.`;
+        } else {
+          options.body = `This notification can display up to ` +
+            `${maxVisibleActions} actions.`;
+        }
+
+        registration.showNotification(title, options);
+
+If / when the user clicks an action button, check the `event.action` value in the
+`noticationclick` event to tell if / which action button was clicked.
+
+`event.action` will contain the `action` value set in the options. In the example about the
+`event.action` values would be one of the following: 'coffee-action', 'doughnut-action',
```

Comment:
In which example? 'In the example above'. 'In the example below.' Something like these would be better.